### PR TITLE
AVX128: More various instructions

### PR DESCRIPTION
**Pull request by Sonicadvance1**
A smattering of scalar operations and vector shifts.

### CODE REVIEW

**Comment by alyssarosenzweig** [comment]
I was going to flag the copypaste in UCOMIS, instead of doing that I wrote #3741 to use 😄

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp
Comment (alyssarosenzweig) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:953]: Unless there's a strong reason otherwise, I'd rather switch on `CompType & 7` which lets you delete 2/3 the code here.

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp
Comment (alyssarosenzweig) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:959]: No need to break after a return (likewise throughout)

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp]: Fixed

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp]: Also fixed

**Comment by Sonicadvance1** [comment]
> I was going to flag the copypaste in UCOMIS, instead of doing that I wrote #3741 to use 😄

Cherry-picked and used.

**Comment by alyssarosenzweig** [comment]
Wrote #3743 for all the Literals flying around here, but I don't mind merging without porting to that, I can do the (fully mechanical) changes after merge if you prefer.

**Comment by Sonicadvance1** [comment]
> Wrote #3743 for all the Literals flying around here, but I don't mind merging without porting to that, I can do the (fully mechanical) changes after merge if you prefer.

Yeesss, that would be good to do in one lump commit once the 66 remaining commits are merged. Since there will be a decent amount of them after the fact and it will be painful to go through each individual commit

**Comment by alyssarosenzweig** [comment]
> > Wrote #3743 for all the Literals flying around here, but I don't mind merging without porting to that, I can do the (fully mechanical) changes after merge if you prefer.
> 
> Yeesss, that would be good to do in one lump commit once the 66 remaining commits are merged. Since there will be a decent amount of them after the fact and it will be painful to go through each individual commit

Sure, that's fine with me. Although https://github.com/torbiak/git-autofixup is basically tailormade for this problem - you do all the changes at the end, run `git autofixup` and then it creates a whole slew of fixup! commits for you to `git rebase` and then it's all fixed up. Magic!

**Comment by alyssarosenzweig** [comment]
I see the fcmp funniness is an existing funniness. So I wrote you #3744

**Comment by alyssarosenzweig** [comment]
VPSRA/VPSLL/VPSRL look like candidates to use the lambda helper. If that doesn't work for some reason (I don't see such a reason but maybe ther'es a subtlelty I'm missing here), we can at least template between the 3

**Comment by alyssarosenzweig** [comment]
oops accidental close

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp
Comment (alyssarosenzweig) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:1367]: `Result = Src`?

**Comment by alyssarosenzweig** [comment]
I just pushed a version with all my comments addressed. I'm happy to merge it in this state if you are. Diff between what I pushed and what you pushed rebased on #3743, for review convenience: https://rosenzweig.io/diff.txt

**Comment by Sonicadvance1** [comment]
> I just pushed a version with all my comments addressed. I'm happy to merge it in this state if you are. Diff between what I pushed and what you pushed rebased on #3743, for review convenience: https://rosenzweig.io/diff.txt

Looks good to me

**Review by alyssarosenzweig — APPROVED** [review]

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp
Comment (lioncash) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:1348]:
```suggestion
  AVX128_ShiftDoubleImm(Op, false /* Left */);
```

Comment (alyssarosenzweig) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:1348]: I think that might be more confusing. it is false that this is Right. it is true that this is Left.

Comment (lioncash) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:1348]: Conversely I find that extremely confusing. The programmatic part should absolutely not be inverting a comment lol, and it genuinely comes across as a copy-paste typo to me. This can be more explicit by just making the comments "Right shift" and "Left shift".

Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:1348]: Easy fix for this is to do a simple enum
```cpp
enum class ShiftDirection {
  RIGHT, LEFT,
};

void AVX128_ShiftDoubleImm(OpcodeArgs, ShiftDirection Dir);

void OpDispatchBuilder::AVX128_VPSRLDQ(OpcodeArgs) {
  AVX128_ShiftDoubleImm(Op, ShiftDirection::RIGHT);
}

void OpDispatchBuilder::AVX128_VPSLLDQ(OpcodeArgs) {
  AVX128_ShiftDoubleImm(Op, ShiftDirection::LEFT);
}
```

Comment (alyssarosenzweig) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp]: We have a winner!

**Pull request merged**
